flow/activities/flowable.go

```diff
@@ -642,47 +657,27 @@ func (a *FlowableActivity) replicateQRepPartition(ctx context.Context,
 	var wg sync.WaitGroup
 
 	var goroutineErr error = nil
-	if config.SourcePeer.Type == protos.DBType_POSTGRES {
-		stream = model.NewQRecordStream(bufferSize)
-		wg.Add(1)
-
-		go func() {
-			pgConn := srcConn.(*connpostgres.PostgresConnector)
-			tmp, err := pgConn.PullQRepRecordStream(ctx, config, partition, stream)
-			numRecords := int64(tmp)
-			if err != nil {
-				a.Alerter.LogFlowError(ctx, config.FlowJobName, err)
-				logger.Error("failed to pull records", slog.Any("error", err))
-				goroutineErr = err
-			} else {
-				err = monitoring.UpdatePullEndTimeAndRowsForPartition(ctx,
-					a.CatalogPool, runUUID, partition, numRecords)
-				if err != nil {
-					logger.Error(err.Error())
-					goroutineErr = err
-				}
-			}
-			wg.Done()
-		}()
-	} else {
-		recordBatch, err := srcConn.PullQRepRecords(ctx, config, partition)
-		if err != nil {
-			a.Alerter.LogFlowError(ctx, config.FlowJobName, err)
-			return fmt.Errorf("failed to pull qrep records: %w", err)
-		}
-		logger.Info(fmt.Sprintf("pulled %d records", len(recordBatch.Records)))
 
-		err = monitoring.UpdatePullEndTimeAndRowsForPartition(ctx, a.CatalogPool, runUUID, partition, int64(len(recordBatch.Records)))
-		if err != nil {
-			return err
-		}
+	stream = model.NewQRecordStream(bufferSize)
+	wg.Add(1)
 
-		stream, err = recordBatch.ToQRecordStream(bufferSize)
+	go func() {
```

Comment:
should split out a preceding PR to make this use errgroup instead of waitgroup+goroutine+goroutineErr

Reply:
https://github.com/PeerDB-io/peerdb/pull/1390